src/renderer/sw_engine/tvgSwRaster.cpp

```diff
@@ -1590,8 +1590,6 @@ static bool _rasterSolidGradientRect(SwSurface* surface, const SwBBox& region, c
 
 static bool _rasterLinearGradientRect(SwSurface* surface, const SwBBox& region, const SwFill* fill)
 {
-    if (fill->linear.len < FLOAT_EPSILON) return false;
```

Comment:
I think the length equation is incorrect. It should be:
- Linear:  skip rendering, if  ((x1 - x2) * (y1 - y2) == 0)
- Radial: skip rendering if  (radius == 0)

Reply:
do you mean `==` or `tvg::zero` ?

Reply:
yes.

Reply:
@mgrudzinska I'm not talking about dx/dy, Skip rendering if the given linear/radial gradient bound is 0.

Reply:
@mgrudzinska I just noticed that two patches should be merged together. They can't exist separately. This looks good to me if [this pull request](https://github.com/thorvg/thorvg/pull/2588) is properly handled altogether. Otherwise, we can consider adding skipping rendering (currently, the radial gradient does this badly as well).